File: SDKs/Aspose.OCR_Cloud_SDK_for_Ruby/lib/aspose_ocr_cloud/api/ocr_api.rb
require "uri"

module AsposeOCRCloud
  class OcrApi
    attr_accessor :api_client

    def initialize(api_client = nil)
      @api_client = api_client || Configuration.api_client
    end

    # Recognize image text from some url if provided or from the request body content, language can be selected, default dictionaries can be used for correction.
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :url The image file url.
    # @option opts [String] :language Language of the document.
    # @option opts [BOOLEAN] :use_default_dictionaries Use default dictionaries for result correction.
    # @option opts [File] :file 
    # @return [OCRResponse]
    def post_ocr_from_url_or_content(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: OcrApi#post_ocr_from_url_or_content ..."
      end
      
      # resource path
      path = "/ocr/recognize".sub('{format}','json')

      # query parameters
      query_params = {}
      query_params[:'url'] = opts[:'url'] if opts[:'url']
      query_params[:'language'] = opts[:'language'] if opts[:'language']
      query_params[:'useDefaultDictionaries'] = opts[:'use_default_dictionaries'] if opts[:'use_default_dictionaries']

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['multipart/form-data']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["file"] = opts[:'file'] if opts[:'file']

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'OCRResponse')
      if Configuration.debugging
        Configuration.logger.debug "API called: OcrApi#post_ocr_from_url_or_content. Result: #{result.inspect}"
      end
      return result
    end

    # Recognize image text, language and text region can be selected, default dictionaries can be used for correction.
    # 
    # @param name Name of the file to recognize.
    # @param [Hash] opts the optional parameters
    # @option opts [String] :language Language of the document.
    # @option opts [Integer] :rect_x Top left point X coordinate of Rectangle to recognize text inside.
    # @option opts [Integer] :rect_y Top left point Y coordinate of Rectangle to recognize text inside.
    # @option opts [Integer] :rect_width Width of Rectangle to recognize text inside.
    # @option opts [Integer] :rect_height Height of Rectangle to recognize text inside.
    # @option opts [BOOLEAN] :use_default_dictionaries Use default dictionaries for result correction.
    # @option opts [String] :storage Image&#39;s storage.
    # @option opts [String] :folder Image&#39;s folder.
    # @return [OCRResponse]
    def get_recognize_document(name, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: OcrApi#get_recognize_document ..."
      end
      
      # verify the required parameter 'name' is set
      fail "Missing the required parameter 'name' when calling get_recognize_document" if name.nil?
      
      # resource path
      path = "/ocr/{name}/recognize".sub('{format}','json').sub('{' + 'name' + '}', name.to_s)

      # query parameters
      query_params = {}
      query_params[:'language'] = opts[:'language'] if opts[:'language']
      query_params[:'rectX'] = opts[:'rect_x'] if opts[:'rect_x']
      query_params[:'rectY'] = opts[:'rect_y'] if opts[:'rect_y']
      query_params[:'rectWidth'] = opts[:'rect_width'] if opts[:'rect_width']
      query_params[:'rectHeight'] = opts[:'rect_height'] if opts[:'rect_height']
      query_params[:'useDefaultDictionaries'] = opts[:'use_default_dictionaries'] if opts[:'use_default_dictionaries']
      query_params[:'storage'] = opts[:'storage'] if opts[:'storage']
      query_params[:'folder'] = opts[:'folder'] if opts[:'folder']

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json', 'text/json', 'application/xml', 'text/xml', 'text/javascript']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'OCRResponse')
      if Configuration.debugging
        Configuration.logger.debug "API called: OcrApi#get_recognize_document. Result: #{result.inspect}"
      end
      return result
    end
  end
end

File: SDKs/Aspose.OCR-Cloud-SDK-for-NodeJS/lib/OcrApi.js
var path = require('path');
var fs = require('fs');
var apiClient = require('./api-client');


function OcrApi(config) {
	var self = this;
	
	apiClient = new apiClient(config);
	this.config = apiClient.config;
}


/**
* PostOcrFromUrlOrContent
* Recognize image text from some url if provided or from the request body content, language can be selected, default dictionaries can be used for correction.
*
* @param  (String) url  -  The image file url. (optional) 
* @param  (String) language  -  Language of the document. (optional) 
* @param  (Boolean) useDefaultDictionaries  -  Use default dictionaries for result correction. (optional) 
* @param  (File) file  -   (required) 
* @returns OCRResponse (Map)
*/
OcrApi.prototype.PostOcrFromUrlOrContent = function(url, language, useDefaultDictionaries, file, callback) {

    var self = this;
	   	
	
	var resourcePath = '/ocr/recognize/?appSid={appSid}&amp;url={url}&amp;language={language}&amp;useDefaultDictionaries={useDefaultDictionaries}';
	
	resourcePath = resourcePath.replace(new RegExp('\\*', 'g'), "");
	resourcePath = resourcePath.replace(new RegExp('&amp;', 'g'), '&');
	resourcePath = resourcePath.replace('&amp;','&').replace("/?","?").replace("toFormat={toFormat}","format={format}").replace("{path}","{Path}");
	
	
	if(typeof url !== 'undefined' &&  url !== null && url!== ''){            
			resourcePath = resourcePath.replace("{" + "url" + "}" , url);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]url.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof language !== 'undefined' &&  language !== null && language!== ''){            
			resourcePath = resourcePath.replace("{" + "language" + "}" , language);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]language.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof useDefaultDictionaries !== 'undefined' &&  useDefaultDictionaries !== null && useDefaultDictionaries!== ''){            
			resourcePath = resourcePath.replace("{" + "useDefaultDictionaries" + "}" , useDefaultDictionaries);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]useDefaultDictionaries.*?(?=&|\\?|$)', 'g'), "");
		}
	
	
	if(this.config.debug){console.log('resourcePath :: ' + resourcePath);}
	
    method = 'POST'
    queryParams = {}
    headerParams = {}
    formParams = {}
    files = {}
	postData = null;
    
	if(file !== 'undefined' && file !== null && file !== ''){
		postData = fs.readFileSync(file)
	}
	
	headerParams['Accept'] = 'application/xml,application/json'
    headerParams['Content-Type'] = 'multipart/form-data'

	return apiClient.InvokeAPI(resourcePath, method, queryParams, postData, headerParams, files, callback);
  }
    
/**
* GetRecognizeDocument
* Recognize image text, language and text region can be selected, default dictionaries can be used for correction.
*
* @param  (String) name  -  Name of the file to recognize. (required) 
* @param  (String) language  -  Language of the document. (optional) 
* @param  (String) rectX  -  Top left point X coordinate of  to recognize text inside. (optional) 
* @param  (String) rectY  -  Top left point Y coordinate of  to recognize text inside. (optional) 
* @param  (String) rectWidth  -  Width of  to recognize text inside. (optional) 
* @param  (String) rectHeight  -  Height of  to recognize text inside. (optional) 
* @param  (Boolean) useDefaultDictionaries  -  Use default dictionaries for result correction. (optional) 
* @param  (String) storage  -  Image&#39;s storage. (optional) 
* @param  (String) folder  -  Image&#39;s folder. (optional) 
* @returns OCRResponse (Map)
*/
OcrApi.prototype.GetRecognizeDocument = function(name, language, rectX, rectY, rectWidth, rectHeight, useDefaultDictionaries, storage, folder, callback) {

    var self = this;
	
	if( typeof name === 'undefined' || name === null && name === ''){
		throw new Error('missing required parameters.');
    }         	
		
	var resourcePath = '/ocr/{name}/recognize/?appSid={appSid}&amp;language={language}&amp;rectX={rectX}&amp;rectY={rectY}&amp;rectWidth={rectWidth}&amp;rectHeight={rectHeight}&amp;useDefaultDictionaries={useDefaultDictionaries}&amp;storage={storage}&amp;folder={folder}';
	
	resourcePath = resourcePath.replace(new RegExp('\\*', 'g'), "");
	resourcePath = resourcePath.replace(new RegExp('&amp;', 'g'), '&');
	resourcePath = resourcePath.replace('&amp;','&').replace("/?","?").replace("toFormat={toFormat}","format={format}").replace("{path}","{Path}");
	
	
	if(typeof name !== 'undefined' &&  name !== null && name!== ''){            
			resourcePath = resourcePath.replace("{" + "name" + "}" , name);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]name.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof language !== 'undefined' &&  language !== null && language!== ''){            
			resourcePath = resourcePath.replace("{" + "language" + "}" , language);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]language.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof rectX !== 'undefined' &&  rectX !== null && rectX!== ''){            
			resourcePath = resourcePath.replace("{" + "rectX" + "}" , rectX);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]rectX.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof rectY !== 'undefined' &&  rectY !== null && rectY!== ''){            
			resourcePath = resourcePath.replace("{" + "rectY" + "}" , rectY);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]rectY.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof rectWidth !== 'undefined' &&  rectWidth !== null && rectWidth!== ''){            
			resourcePath = resourcePath.replace("{" + "rectWidth" + "}" , rectWidth);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]rectWidth.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof rectHeight !== 'undefined' &&  rectHeight !== null && rectHeight!== ''){            
			resourcePath = resourcePath.replace("{" + "rectHeight" + "}" , rectHeight);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]rectHeight.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof useDefaultDictionaries !== 'undefined' &&  useDefaultDictionaries !== null && useDefaultDictionaries!== ''){            
			resourcePath = resourcePath.replace("{" + "useDefaultDictionaries" + "}" , useDefaultDictionaries);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]useDefaultDictionaries.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof storage !== 'undefined' &&  storage !== null && storage!== ''){            
			resourcePath = resourcePath.replace("{" + "storage" + "}" , storage);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]storage.*?(?=&|\\?|$)', 'g'), "");
		}
	
	if(typeof folder !== 'undefined' &&  folder !== null && folder!== ''){            
			resourcePath = resourcePath.replace("{" + "folder" + "}" , folder);
	}else{
		resourcePath = resourcePath.replace(new RegExp('[&?]folder.*?(?=&|\\?|$)', 'g'), "");
		}
	
	
	if(this.config.debug){console.log('resourcePath :: ' + resourcePath);}
	
    method = 'GET'
    queryParams = {}
    headerParams = {}
    formParams = {}
    files = {}
	postData = null;
    
	
	headerParams['Accept'] = 'application/xml,application/json'
    headerParams['Content-Type'] = 'application/json'

	return apiClient.InvokeAPI(resourcePath, method, queryParams, postData, headerParams, files, callback);
  }
    

module.exports = OcrApi


